src/handlers/RequestHandlers.ts

```
@@ -72,175 +75,92 @@ function setupToolHandlers(server: Server, core: MemoryPickleCore): void {
-        if (!fileName || fileName.trim() === '') {
-          throw new Error(`Invalid resource URI: filename is empty or missing. URI: ${request.params.uri}`);
-        }
 
-        console.error(`Debug: Processed fileName: "${fileName}"`);
+      if (url.protocol === 'memory:') {
+        const resourcePath = url.pathname;
+
+        switch (resourcePath) {
+          case '/current-session':
+            // Return current in-memory database as JSON
+            const database = core.getDatabase();
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)